cluster/src/table_manager.rs

```diff
@@ -5,67 +5,80 @@ use std::{
     sync::RwLock,
 };
 
-use meta_client_v2::{SchemaId, ShardId, ShardInfo, ShardTables, TableId, TableInfo};
+use meta_client_v2::types::{
+    CreateTableCmd, SchemaId, ShardId, ShardInfo, ShardTables, TableId, TableInfo,
+};
+use snafu::OptionExt;
+
+use crate::{Result, ShardNotFound};
 
-use super::Result;
-use crate::ShardNotFound;
+pub type TableName = String;
+pub type SchemaName = String;
 
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
 struct SchemaInfo {
-    name: String,
+    name: SchemaName,
     id: SchemaId,
 }
 
-pub struct TableManager {
-    inner: RwLock<TableManagerInner>,
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct ShardTableInfo {
+    pub shard_id: ShardId,
+    pub table_info: TableInfo,
 }
 
-impl TableManager {
-    pub fn new() -> Self {
-        Self {
-            inner: RwLock::new(TableManagerInner {
-                shards_info: Vec::new(),
-                schemas_info: HashMap::new(),
-                tables: BTreeMap::new(),
-            }),
+impl From<&CreateTableCmd> for ShardTableInfo {
+    fn from(cmd: &CreateTableCmd) -> Self {
+        let table_info = TableInfo {
+            id: cmd.id,
+            name: cmd.name.clone(),
+            schema_id: cmd.schema_id,
+            schema_name: cmd.schema_name.clone(),
+        };
+        ShardTableInfo {
+            shard_id: cmd.shard_id,
+            table_info,
         }
     }
+}
+
+/// TableManager manages information about tables, shards, schemas and their
+/// relationships:
+/// * one shard -> multiple tables
+/// * one schema -> multiple tables
+#[derive(Debug, Default)]
+pub struct TableManager {
+    inner: RwLock<Inner>,
+}
 
-    pub fn get_shards_info(&self) -> Vec<ShardInfo> {
-        self.inner.read().unwrap().get_shards_info()
+impl TableManager {
```

Comment:
Do you think we need an interface like `get_shard_tables` or `get_tables_by_shard`?

Reply:
Yes, such a method is necessary but we can add it in the future PR.